client/components/route/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+const RouteContext = React.createContext( {
+	currentSection: null,
+	currentRoute: '',
+	currentQuery: false,
+} );
+
+export function RouteProvider( {
+	currentSection = null,
```

Comment:
There's a subtle difference between `null` and `false`. Trying to access a property of `null` throws a runtime error. On the other hand, `false.foo` simply returns `undefined` -- `false` is an object! Also, for example, `false.toString()` returns a `"false"` string.

Changing `false` to `null` is therefore a bit dangerous: if any of existing code calls `currentSection.group`, for example, it can break.

I'll update the `currentSection` default back to `false`.

Reply:
Thank you for the detailed explanation — do you think it'd be valuable to add it as a comment? Otherwise, it may seem weird that `currentSection` can be either a Boolean or a String